### PR TITLE
feat(file-upload): implement component

### PR DESCRIPTION
**Pull request by dpellier**

### CODE REVIEW

**Review by feoche — CHANGES_REQUESTED** [review]
## Style
- [ ] When having global error, a Divider should be displayed
- [ ] When having global error, border should be in red color

File: packages/storybook/stories/components/file-upload/documentation.mdx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import * as FileUploadStories from './file-upload.stories';
+import { LINK_ID } from '../../zeroheight';
+
+<Meta of={ FileUploadStories } name='Documentation' />
+
+<Banner of={ FileUploadStories } />
+
+# Overview
+
+File-upload allow users to select one or more files to upload to a specific location:
```
Comment (Leotheluck) [inline packages/storybook/stories/components/file-upload/documentation.mdx:14]: allows*

File: packages/storybook/stories/components/file-upload/documentation.mdx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+<Canvas of={ FileUploadStories.Overview } sourceState='none' />
+
+<DocNavigator of={ FileUploadStories } linkId={ LINK_ID.FILE_UPLOAD } />
+
+<Markdown>{ SpecificationsFileUpload }</Markdown>
+
+# How to use?
+
+The component does not handle any actual upload, nor it knows the status of each file (regarding progress or error),
```
Comment (Leotheluck) [inline packages/storybook/stories/components/file-upload/documentation.mdx:24]: The component does not actually handle any upload, nor does it know the status of each file*

File: packages/storybook/stories/components/file-upload/documentation.mdx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+as the upload process is different for every integration.
+
+All the files you'll receive from this component are typed as `OdsFile`.
+
+This is an extend of the native `File` interface with optional attribute that you can set to alter the display:
```
Comment (Leotheluck) [inline packages/storybook/stories/components/file-upload/documentation.mdx:29]: extension*

File: packages/storybook/stories/components/file-upload/documentation.mdx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+- `error`: set this if the upload of the file fail, to display an error message near it on the component file list.
```
Comment (Leotheluck) [inline packages/storybook/stories/components/file-upload/documentation.mdx:30]: set this if the upload of the file fails, in order to display an error message below the file list.*

**Review by Leotheluck — COMMENTED** [review]
## General
- [ ] Should we remove the line "Accepted files:" if nothing is specified?

## Migration guide
- [ ] Should we remove the migration guide page as there is no migration possible?

**Review by feoche — CHANGES_REQUESTED** [review]
## Style
- [x] Global error should be left-aligned

## Demo
- [ ] Add in `accepted` description : "Bound to native file upload behavior. For label, use `acceptedFileLabel` attribute"

**Review by feoche — APPROVED** [review]

**Review by astagnol — APPROVED** [review]

**Review by aesteves60 — APPROVED** [review]

**Review by feoche — APPROVED** [review]

**Pull request merged**